src/App.tsx

```diff
@@ -1,26 +1,174 @@
+    () => todos.filter(todo => !todo.completed).length,
+    [todos],
+  );
+
+  const onAddTodo = async (todoTitle: string) => {
+    setTempTodo({ id: 0, title: todoTitle, completed: false, userId: USER_ID });
+    try {
+      const newTodo = await addTodo({ title: todoTitle, completed: false });
+
+      setTodos(prev => [...prev, newTodo]);
+    } catch (err) {
+      setErrorMessage(ErrorType.AddTodo);
+      inputAddRef?.current?.focus();
+      throw err;
+    } finally {
+      setTempTodo(null);
+    }
+  };
+
+  const onRemoveTodo = async (todoId: number) => {
+    setLoadingTodoIds(prev => [...prev, todoId]);
+    try {
+      await deleteTodo(todoId);
+      setTodos(prev => prev.filter(todo => todo.id !== todoId));
+    } catch (err) {
+      setErrorMessage(ErrorType.DeleteTodo);
+      inputAddRef?.current?.focus();
+      throw err;
+    } finally {
+      setLoadingTodoIds(prev => prev.filter(id => id !== todoId));
+    }
+  };
+
+  const onUpdateTodo = async (todoToUpdate: Todo) => {
+    setLoadingTodoIds(prev => [...prev, todoToUpdate.id]);
+    try {
+      const updatedTodo = await updateTodo(todoToUpdate);
+
+      setTodos(prev =>
+        prev.map(todo => (todo.id === updatedTodo.id ? updatedTodo : todo)),
+      );
+    } catch (err) {
+      setErrorMessage(ErrorType.UpdateTodo);
+      throw err;
+    } finally {
+      setLoadingTodoIds(prev => prev.filter(id => id !== todoToUpdate.id));
+    }
+  };
+
+  const onClearCompleted = async () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    completedTodos.forEach(todo => {
+      onRemoveTodo(todo.id);
+    });
+  };
+
+  const onToggleAll = async () => {
+    if (todosActiveNum > 0) {
+      const activeTodos = todos.filter(todo => !todo.completed);
+
+      activeTodos.forEach(todo => {
+        onUpdateTodo({ ...todo, completed: true });
+      });
+    } else {
+      todos.forEach(todo => {
+        onUpdateTodo({ ...todo, completed: false });
+      });
+    }
+  };
+
+  const inputAddRef = useRef<HTMLInputElement>(null);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <TodoHeader
+          onAddTodo={onAddTodo}
+          setErrorMessage={setErrorMessage}
+          isInputDisabled={!!tempTodo}
+          inputRef={inputAddRef}
+          onToggleAll={onToggleAll}
+          areAllTodosCompleated={areAllTodosCompleated}
+          todosLength={todos.length}
+        />
+
+        {(todos.length > 0 || tempTodo) && (
```

Comment:
You also can use this syntax 
-        {todos.length > 0 && (
+        {!!todos.length && (